core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -1424,33 +1423,47 @@ protected void bitcoinSerializeToStream(OutputStream stream) throws IOException
     }
 
     /**
-     * Serialize according to <a href="https://github.com/bitcoin/bips/blob/master/bip-0144.mediawiki">BIP144</a> or the
-     * <a href="https://en.bitcoin.it/wiki/Protocol_documentation#tx">classic format</a>, depending on if segwit is
-     * desired.
+     * Write this transaction into the given buffer.
+     *
+     * @param buf buffer to write into
+     * @return the buffer
+     * @throws BufferOverflowException if the transaction doesn't fit the remaining buffer
      */
-    protected void bitcoinSerializeToStream(OutputStream stream, boolean useSegwit) throws IOException {
+    protected ByteBuffer write(ByteBuffer buf, boolean useSegwit) throws BufferOverflowException {
         // version
-        writeInt32LE(version, stream);
+        ByteUtils.writeInt32LE(version, buf);
         // marker, flag
         if (useSegwit) {
-            stream.write(0);
-            stream.write(1);
+            buf.put((byte) 0);
+            buf.put((byte) 1);
         }
         // txin_count, txins
-        stream.write(VarInt.of(inputs.size()).serialize());
+        VarInt.of(inputs.size()).write(buf);
         for (TransactionInput in : inputs)
-            stream.write(in.serialize());
+            in.write(buf);
         // txout_count, txouts
-        stream.write(VarInt.of(outputs.size()).serialize());
+        VarInt.of(outputs.size()).write(buf);
         for (TransactionOutput out : outputs)
-            stream.write(out.serialize());
+            out.write(buf);
         // script_witnisses
         if (useSegwit) {
             for (TransactionInput in : inputs)
-                stream.write(in.getWitness().serialize());
+                in.getWitness().write(buf);
         }
         // lock_time
-        writeInt32LE(vLockTime.rawValue(), stream);
+        ByteUtils.writeInt32LE(vLockTime.rawValue(), buf);
+        return buf;
+    }
+
+    /**
+     * Serialize according to <a href="https://github.com/bitcoin/bips/blob/master/bip-0144.mediawiki">BIP144</a> or the
```

Comment:
This comment should probably be moved up to the ByteBuffer-based `write` which actually implements the BIP.

Reply:
Changed: https://github.com/bitcoinj/bitcoinj/compare/79b8257961f637276473a2b82d6738ff8baffdc5..e20e5948dd9714e5160815b2d36f2a789ca7ea1f